### ISSUE TITLE
Training utilities

### ISSUE THREAD

**Issue by smguo** [czbiohub/dynamorph#12]
Would be useful to add training utilities that  
- split the data into train, validation, test for monitoring overfitting and tuning hyper parameters
- at the end of every epoch, save model if the validation loss improves. Right now the model only gets saved at the end of training. 
- early stopping training if validation loss does not improve

**Comment by mattersoflight** [comment]
These are reconstruction losses observed during previous training. 
"Reconstruction loss averaged over all training cell patches is 0.16 ± 0.08SD after normalization of both channels. This was further validated on the test dataset, on which model reconstruction loss is 0.18 ± 0.07SD"

<img width="923" alt="image" src="https://user-images.githubusercontent.com/2934183/106699573-fdf39c00-6597-11eb-9a68-2f2941c8f2a2.png">

**Comment by smguo** [comment]
addressed in #22
